Nos/Views/Modifiers/ReportMenuModifier.swift

```diff
@@ -173,16 +196,39 @@ struct ReportMenuModifier: ViewModifier {
         }
     }
     
-    func mute(author: Author) {
-        Task {
-            do {
-                try await author.mute(viewContext: viewContext)
-            } catch {
-                Log.error(error.localizedDescription)
+    func mute(author: Author) async throws {
+        do {
+            try await author.mute(viewContext: viewContext)
+        } catch {
+            Log.error(error.localizedDescription)
+            throw error
+        }
+    }
+
+    func unmute(author: Author) async throws {
+        do {
+            try await author.unmute(viewContext: viewContext)
+        } catch {
+            Log.error(error.localizedDescription)
+            throw error
+        }
+    }
+
+    /// Determines the visibility status for a flagged account and applies the appropriate action.
+    private func determineFlaggedAccoutVisibility() async {
+        if let author = reportedObject.author {
+            guard case .visibility(let visibilityCategory) = selectedVisibilityOption?.category else { return }
+
+            if visibilityCategory == .mute {
+                guard !author.muted else { return }
+                try? await mute(author: author)
+            } else {
+                guard author.muted else { return }
+                try? await unmute(author: author)
```

Comment:
I'm not fully understanding when we'd unmute a user in the report/flag flow. What happens that leads to this?

Reply:
In the new flow, we have an extra section to either mute or unmute a user. In a user where the user has selected the mute option before for a user, they still have a choice to unmute the user again when flagging the user another time.

https://www.figma.com/design/s0qf4VmyQygydP8MIQazZc/Nos?node-id=8712-19151&t=vv2ep2sqTLOMCY0Y-4